n3fit/src/n3fit/model_trainer.py

```diff
@@ -268,7 +267,7 @@ def _fill_the_dictionaries(self):
             - ``name``: names of the experiment
             - ``ndata``: number of experimental points
         """
-        for exp_dict in self.exp_info:
+        for index, exp_dict in enumerate(self.exp_info[0]):
```

Comment:
Why do you need the index and the enumerate here?

Reply:
Because essentially we need to transpose the whole data structure of exp_info to stack the replica experimental masked data. I agree it feels ad-hoc inside the loop. I can create a new data structure and fill it outside the loop.

Reply:
Maybe I've skipped something, but it is not being used, is it?

Reply:
Done in [febbfd4](https://github.com/NNPDF/nnpdf/pull/1788/commits/febbfd44a7b514469e0b14acc71381e01704e02f)